Controllers/AdminController.cs

```diff
@@ -77,11 +114,15 @@ public IActionResult CreateBook(CreateBookVM bookVM, long id)
          public IActionResult UpdateBook(long id)
         {
             //TODO: Display the database info for the Book (and its Author) indicated by the id 
-            Book book;
+            Book book = _db.Books.Include(b => b.Author).SingleOrDefault(b => b.Id == id);
 
             var createbookVM = new CreateBookVM
             {
-               
+                BookId = book.Id,
+                Title = book.Title,
+                SKU = book.SKU,
+                Price = book.Price,
+                Writers = _db.Writers.Where(w => w.Id == book.Author.Id)
             };
             return View("CreateBook", createbookVM);
```

Comment:
You don't need to repeat this code, just direct the `Update` link to  reuse the Create book method with the `asp-route-id=book.Id`